### PR TITLE
NikitK/ Refactoring functions inside common and hook folders to typescript

### PR DESCRIPTION
**Pull request by NikitK-deriv**
Changes:

Were add types to common utils and custom hooks. Unused functions was removed. 
Hook `useStateWithCallback` has removed due that behavior can be reached much easier (in accordion.tsx)
Hook `useCookieState` and function `getDateFromToday` was pass to hook `useWebsiteStatus` due they are not used nowhere else

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/common/country-base.ts [hunk 1]
```diff
@@ -319,9 +319,10 @@ export const african_countries = [
     'zm',
     'zw',
 ]
-export const isEuCountry = (clients_country) => eu_countries.includes(clients_country)
+export const isEuCountry = (clients_country: string): boolean =>
+    eu_countries.includes(clients_country)
```
Comment (michio-rework) [inline src/common/country-base.ts:323]:
```suggestion

export const eu_countries = [
    'it',
    'de',
    'fr',
    'lu',
    'gr',
    'mf',
    'es',
    'sk',
    'lt',
    'nl',
    'at',
    'bg',
    'si',
    'cy',
    'be',
    'ro',
    'hr',
    'pt',
    'pl',
    'lv',
    'ee',
    'cz',
    'fi',
    'hu',
    'dk',
    'se',
    'ie',
    'im',
    'gb',
    'mt',
] as const;
type TEuCountries = typeof eu_countries[number];

export const isEuCountry = (clients_country: TEuCountries): boolean =>
    eu_countries.includes(clients_country)

```
how about this approach? 
we get more type safty and cool features of typescript. 
let me know what you think, thanks

File: src/common/login.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ const Login = (() => {
             : `${oauth_url}/oauth2/authorize?app_id=${deriv_app_id}&l=${language}&brand=${brand_name.toLowerCase()}${affiliate_token_link}${cookies_link}&platform=${sub_url}`
     }
 
-    const initOneAll = (provider) => {
+    const initOneAll = (provider: string): void => {
         const social_login_url = `${loginUrl()}&social_signup=${provider}`
```
Comment (michio-rework) [inline src/common/login.ts:38]:
```suggestion
type TSocialProvider = "google" | "facebook" | "apple" 
const initOneAll = (provider: TSocialProvider) => {}
```
As far as I could find in our codebase these are the only possible values, let me know what you think about this ,thanks

File: src/components/hooks/use-open-live-chat-redirection.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import { redirectOpenLiveChatBox } from 'common/utility'
 
-export const useOpenLiveChat = (props) => {
+export const useOpenLiveChat = (props?: boolean) => {
```
Comment (michio-rework) [inline src/components/hooks/use-open-live-chat-redirection.tsx:4]: how about we change `props` to `isRedirectEnabled` or something else, more meaningful name I mean?

**Review by michio-rework — COMMENTED** [review]
Left Some comments

File: src/common/country-base.ts (hunk 1 above)
Comment (NikitK-deriv) [reply src/common/country-base.ts]: I exactly prefer your decision. Thanks for share, very smart approach 👍

File: src/common/login.ts (hunk 2 above)
Comment (NikitK-deriv) [reply src/common/login.ts]: Thanks a lot. I'll check other functions to follow this approach 🙇

File: src/components/hooks/use-open-live-chat-redirection.tsx (hunk 3 above)
Comment (NikitK-deriv) [reply src/components/hooks/use-open-live-chat-redirection.tsx]: Have sense, changed on `is_triggered`

File: src/common/country-base.ts [hunk 4]
```diff
@@ -318,10 +316,14 @@ export const african_countries = [
...
+export const isEuCountry = (clients_country: TEuCountries): boolean =>
+    eu_countries.includes(clients_country)
 
-export const isP2PAllowedCountry = (clients_country) =>
+type TP2pCountries = typeof p2p_allowed_countries[number]
+export const isP2PAllowedCountry = (clients_country: TP2pCountries): boolean =>
     p2p_allowed_countries.includes(clients_country)
 
-export const isUK = (clients_country) => clients_country === 'gb'
+export const isUK = (clients_country: 'gb'): boolean => clients_country === 'gb'
```
Comment (maryia-binary) [inline src/common/country-base.ts:329]:
```suggestion
export const isUK = (clients_country: string): boolean => clients_country === 'gb'
```

Comment (NikitK-deriv) [reply src/common/country-base.ts]: Resolved

**Comment by eduard-deriv** [comment]
after fixes according to the comments, LGTM.

**Review by eduard-deriv — APPROVED** [review]

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**